src/app/profile/profile.component.js

```diff
@@ -0,0 +1,385 @@
+        });
+      }
+      if(item.self && item.delete == undefined) { // update existing phone number
+        this.profileService.updatePhoneNumber(item)
+          .subscribe(
+            () => {
+              this.resetPhoneNumberForms();
+              this.success = true;
+              this.phonesLoading = false;
+            },
+            error => {
+              this.phoneNumberError = 'Failed updating phone numbers.';
+              this.$log.error(this.phoneNumberError, error.data);
+              this.phonesLoading = false;
+            }
+          );
+      } else if(item.self && item.delete) { // delete existing phone number
+        this.profileService.deletePhoneNumber(item)
+          .subscribe(
+            () => {
+              pull(this.phoneNumbers, item);
+              this.resetPhoneNumberForms();
+              this.success = true;
+              this.phonesLoading = false;
+            },
+            error => {
+              this.phoneNumberError = 'Failed deleting phone numbers.';
+              this.$log.error(this.phoneNumberError, error.data);
+              this.phonesLoading = false;
+            }
+          );
+      } else if(!item.self && !item.delete){ // add new phone number
+        this.profileService.addPhoneNumber(item)
+          .subscribe(
+            data => {
+              data.spouse = item.spouse;
+              this.resetPhoneNumberForms();
+              this.success = true;
+              item['phone-number'] = data['phone-number'];
+              this.phonesLoading = false;
+            },
+            error => {
+              this.phoneNumberError = 'Failed adding phone numbers.';
+              this.$log.error(this.phoneNumberError, error.data);
+              this.phonesLoading = false;
+            }
+          );
+      }
+    }
+  }
+
+  deletePhoneNumber(phone, index) {
+    phone.delete = true;
+    if(phone.self) { // set existing phone number for a deletion
+      this.phoneNumberForms[index].$setDirty();
+    } else { // reset validations
+      this.phoneNumberForms[index].$setValidity();
+      this.phoneNumberForms[index].$setPristine();
+    }
+  }
+
+  invalidPhoneNumbers() {
+    let hasInvalid = false;
+    angular.forEach(this.phoneNumberForms, (form) => {
+      if(form && form.$invalid) {
+        hasInvalid = true;
+        return;
+      }
+    });
+    return hasInvalid;
+  }
+
+  dirtyPhoneNumbers() {
+    let hasDirty = false;
+    angular.forEach(this.phoneNumberForms, (form) => {
+      if(form && form.$dirty) {
+        hasDirty = true;
+        return;
+      }
+    });
+    return hasDirty;
+  }
+
+  resetPhoneNumberForms() {
+    angular.forEach(this.phoneNumberForms, (form) => {
+      if(form)form.$setPristine();
+    });
+  }
+
+  loadMailingAddress() {
+    this.mailingAddressLoading = true;
+    this.profileService.getMailingAddress()
+      .subscribe(
+        data => {
+          data.address = formatAddressForTemplate(data.address);
```

Comment:
Can you move this into the service? I think that's how the rest of the services have been implemented.